src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -4323,6 +4323,8 @@ public Type boxedTypeOrType(Type t) {
      * Return the primitive type corresponding to a boxed type.
      */
     public Type unboxedType(Type t) {
+        if (t.hasTag(ERROR))
```

Comment:
This is not necessary, right? After all, you check for errors upfront, in the conditional code. Reason I'm asking is that it's not clear to me as to whether we should just return the error type, or no type. Also, this might affect code which is unrelated to the one you are fixing.

Reply:
(I am not an expert on types, so I may be wrong.)

This method seems to return either the primitive type for which `t` is a box, or `noType`. So returning a `noType` for erroneous type seems reasonable to me. Basically, it would mean that an erroneous type is not a box for any type, which seems sensible. (Currently, every erroneous type is seen as the box for the first primitive type in the sequence, which I think is `byte`.)

I agree only one of these two changes is needed, although I'd personally probably try to go with the one here, in `unboxedType`, as it seems like a generally desirable behavior to me. I agree it can have effects on other parts of the code, though, so it has more potential to break something.

Reply:
I'm ok with going for the proposed fix, as long as tests are ok. Approving.

Reply:
Yes, this is not necessary if we only solve this bug. But it is useful logically. And all the related tests passed locally and passed at the `Pre-submit tests`. So I think it is good to remain it.